### ISSUE TITLE
[Pay 4/2][$500] The employee can edit description and category of paid request by deep link

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#37910]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.48-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @DylanDylann 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1709798205710159

## Action Performed:
1. As a employee, create a request
2. Go to admin account and pay the above request
3. As a admin, go to detail request paid and click edit category
4. Copy URL
5. Access this URL in the employee account

## Expected Result: 
The not found page should be displayed

## Actual Result: 
The category page is displayed

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/16d31195-04af-4440-8ca7-431321e1b641

https://github.com/Expensify/App/assets/38435837/29f442e6-c8e4-4e32-b5bc-8778042107e8

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012583704b4cee619b</li>
        <li>Upwork Job ID: 1766956092089245696</li>
        <li>Last Price Increase: 2024-03-10</li>
    </ul>
</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
We don't check if request is paid or not 

### What changes do you think we should make in order to solve the problem?

In here
https://github.com/Expensify/App/blob/c270b5e233c61f833fbfbc83498724f5496a7766/src/pages/iou/request/step/IOURequestStepCategory.js#L80
We should add condition to check if 
```
const shouldShowCategory = ReportUtils.isGroupPolicy(report) && (transactionCategory || OptionsListUtils.hasEnabledOptions(_.values(policyCategories))

const shouldShowNotFoundPage = !shouldShowCategory || (isEditing && (isSplitBill ? !canEditSplitBill : !canEditMoneyRequest(reportAction)));
```

In the description page, we should add the new variable called shouldShowNotFoundPage and pass it to StepScreenWrapper
```
const shouldShowNotFoundPage = isEditing && (isSplitBill ? !canEditSplitBill : !canEditMoneyRequest(reportAction));
```
### What alternative solutions did you explore? (Optional)

**Comment by DylanDylann** [comment]
@mallenexpensify  While reviewing another PR, I saw this bug and reported it. I already have the context of this bug. Could I take over this bug as C+ role if this issue is external?

**Comment by Victor-Nyagudi** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The employee can edit the description and category of a paid request via a deep link. 

I can't reproduce the bug with the category as of this posting, but it exists with editing the description via deep link. Regardless, a similar approach can be utilized. 

### What is the root cause of that problem?

`IOURequestStepDescription` doesn't have the logic to show the 'Not Found' page, and the `shouldShowNotFoundPage` prop is missing in `StepScreenWrapper`, so it defaults to `false`. 

https://github.com/Expensify/App/blob/537b7c081392c3721293989a9b6517f698580607/src/pages/iou/request/step/IOURequestStepDescription.js#L145-L150

### What changes do you think we should make in order to solve the problem?

I propose following a similar approach to [`IOURequestStepTag`](https://github.com/Expensify/App/blob/537b7c081392c3721293989a9b6517f698580607/src/pages/iou/request/step/IOURequestStepTag.js#L88) where we check if the user is editing a money request and if they are allowed to edit the money request to determine whether or not to show the 'Not Found' page. 

1. Add `report` and `parentReportActions` props to `IOURequestStepDescription`.

```diff
function IOURequestStepDescription({
    route: {
        params: {action, iouType, reportID, backTo},
    },
    transaction,
    splitDraftTransaction,
    policy,
    policyTags,
    policyCategories,
+   parentReportActions,
+   report,
}) 
``` 

2. Check if `isEditing` is `true` and `canEditMoneyRequest` is `false` to determine if we should show the 'Not Found' page. The following is added before the `return` statement in `IOURequestStepDescription`.

```javascript
const isEditing = action === CONST.IOU.ACTION.EDIT;

const parentReportAction = parentReportActions[report.parentReportActionID];

const shouldShowNotFoundPage = isEditing && !canEditMoneyRequest(parentReportAction);
```

3. Add `shouldShowNotFoundPage` prop to `StepScreenWrapper`.

```diff
<StepScreenWrapper
    headerTitle={translate('common.description')}
    onBackButtonPress={navigateBack}
    shouldShowWrapper
    testID={IOURequestStepDescription.displayName}
+   shouldShowNotFoundPage={shouldShowNotFoundPage}
>
```

4. Add the following inside `withOnyx` to get the `parentReportActions`, similar to what's in `IOURequestStepTag`.

```javascript
parentReportActions: {
    key: ({report}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report ? report.parentReportID : '0'}`,
    canEvict: false,
},
```

### What alternative solutions did you explore? (Optional)

None so far.

**Comment by mallenexpensify** [comment]
Yes @DylanDylann , I assigned you as C+.  Please review the above proposals, thx

**Comment by DylanDylann** [comment]
@nkdengineer's [proposal](https://github.com/Expensify/App/issues/37910#issuecomment-1984056331) looks good to me. 

🎀 👀 🎀 C+ reviewed

**Comment by DylanDylann** [comment]
@mallenexpensify The PR is deployed to production 2 days ago. Please help to update this issue

**Comment by mallenexpensify** [comment]
Thanks for the ping @DylanDylann , this coulda got stuck in `reviewing` and not going overdue.  Updated.

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


**Regression Test Proposal**

1. As a employee, create a request
2. Go to admin account and pay the above request
3. As a admin, go to detail request paid and click edit category
4. Copy URL
5. Access this URL in the employee account
6. The not found page should be displayed


Do we agree 👍 or 👎

**Comment by mallenexpensify** [comment]
@DylanDylann , @nkdengineer , can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~012583704b4cee619b
